### ISSUE TITLE
Type overrides type

### ISSUE THREAD

**Issue by OriWritesAPlugin** [eerussianguy/Entity-Zoning-Supervisor#1]
Consulting the documentation on spawn restrictions, this is the generic form:

{
  "type": "ez_supervisor:distance_below_sea_level",
  "distance": 5
}

Wanting to add a restriction that a mob can only spawn from a spawner, I would want to use ez_supervisor:spawn_type:

{
  "type": "ez_supervisor:spawn_type",
}

Checking the docs, the additional parameter would be "type":

{
  "type": "ez_supervisor:spawn_type",
  "type": "spawner"
}

but since this is JSON, I'd expect those two "type" keys to collide. Checking the code, it does look like "type" is expected:

https://github.com/eerussianguy/Entity-Zoning-Supervisor/blob/56b6e9fd01c3064565c21c97493be36b40a65bd3/src/main/java/com/eerussianguy/ez_supervisor/common/data/SpawnRestrictionTypes.java#L76

Basically, if I've got the big stupid right now, I'd request an example on using spawn_type. If not, maybe "type"->"reason" or something.

Mod's been fantastic otherwise :]

**Comment by eerussianguy** [comment]
fixed
